linera-execution/src/execution.rs

```diff
@@ -347,41 +347,14 @@ where
         txn_tracker: &mut TransactionTracker,
     ) -> Result<(), ExecutionError> {
         assert_eq!(context.chain_id, self.context().extra().chain_id());
-        match message {
-            Message::System(message) => {
-                let mut outcome = RawExecutionOutcome {
-                    authenticated_signer: context.authenticated_signer,
-                    refund_grant_to: context.refund_grant_to,
-                    ..Default::default()
-                };
-                outcome.messages.push(RawOutgoingMessage {
-                    destination: Destination::Recipient(context.message_id.chain_id),
-                    authenticated: true,
-                    grant,
-                    kind: MessageKind::Bouncing,
-                    message,
-                });
-                txn_tracker.add_system_outcome(outcome)?;
-            }
-            Message::User {
-                application_id,
-                bytes,
-            } => {
-                let mut outcome = RawExecutionOutcome {
-                    authenticated_signer: context.authenticated_signer,
-                    refund_grant_to: context.refund_grant_to,
-                    ..Default::default()
-                };
-                outcome.messages.push(RawOutgoingMessage {
-                    destination: Destination::Recipient(context.message_id.chain_id),
-                    authenticated: true,
-                    grant,
-                    kind: MessageKind::Bouncing,
-                    message: bytes,
-                });
-                txn_tracker.add_user_outcome(application_id, outcome)?;
-            }
-        }
+        txn_tracker.add_outgoing_message(OutgoingMessage {
+            destination: Destination::Recipient(context.message_id.chain_id),
+            authenticated_signer: context.authenticated_signer,
+            refund_grant_to: context.refund_grant_to.filter(|_| !grant.is_zero()),
```

Comment:
I checked that we ignore the `refund_grant_to` if `grant == 0`.

Reply:
Yes, but it doesn't hurt to already omit it from the messages and save a few bytes.